x/gamm/pool-models/stableswap/README.md

```diff
@@ -1,4 +1,106 @@
-# Stableswap
+# Solidly Stableswap
+
+Stableswaps are pools that offer low slippage for two assets that are intended to be tightly correlated.
+There is a price ratio they are expected to be at, and the AMM offers low slippage around this price.
+There is still price impact for each trade, and as the liquidity becomes more lop-sided, the slippage drastically increases.
 
 This package implements the Solidly stableswap curve, namely a CFMM with
-invariant: `xy(x^2 + y^2) = k`
+invariant: $f(x, y) = xy(x^2 + y^2) = k$
+
+It is generalized to the multi-asset setting as $f(a_1, ..., a_n) = a_1 * ... * a_n (a_1^2 + ... + a_n^2)$
+
+## Choice of curve 
+
+{TODO: Include some high level summary of the curve}
+
+## Pool configuration
+
+{include details of scaling factors and how they imply price peg, scaling factor governors}
+
+## Algorithm details
+
+The AMM pool interfaces requires implementing the following stateful methods:
+
+```golang
+	SwapOutAmtGivenIn(tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.Coin, err error)
+	SwapInAmtGivenOut(tokenOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error)
+
+    SpotPrice(baseAssetDenom string, quoteAssetDenom string) (sdk.Dec, error)
+
+	JoinPool(tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, err error)
+	JoinPoolNoSwap(tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, err error)
+	ExitPool(numShares sdk.Int, exitFee sdk.Dec) (exitedCoins sdk.Coins, err error)
+```
+
+The "constant" part of CFMM's imply that we can reason about all their necessary algorithms from just the CFMM equation. There are still multiple ways to solve each method. We detail below the ways in which we do so. This is organized by first discussing variable substitutions we do, to be in a more amenable form, and then the details of how we implement each method.
+
+<!---  This is true, but however the details of this curve actually lend itself better to using more general solutions for swaps via binary search. (Inspired from https://arxiv.org/abs/2111.13740 ) The remainder of the methods we implement via generic CFMM ideas.
+
+We detail below both what the direct solution via the CFMM equation is, and the solution we use.
+Then we show how these are used to give all of the swqp equations. --->
+
+### CFMM function
+
+Most operations we do only need to reason about two of the assets in a pool, and sometimes only one.
+We wish to have a simpler CFMM function to work with in these cases.
+Due to the CFMM equation $f$ being a symmetric function, we can wlog reorder the arguments to the function. Thus we put the assets of relevance at the beginning of the function. So if two assets $x, y$, we write: $f(x,y, a_3, ... a_n) = xy * a_3 * ... a_n (x^2 + y^2 + a_3^2 + ... + a_n^2)$.
+
+We then take a more convenient expression to work with, via variable substition.
+$$ \begin{equation}
```

Comment:
This isn't rendering for me

Reply:
Hrmm, this is official mathjax

I think we just link to something else for rendering then, I don't want to go through the trouble of replacing these with images. Or we can fix rendering in a second pass later. It should work in your IDE + hackmd